src/advanced/JupyterNotebook/index.tsx

```diff
@@ -0,0 +1,322 @@
+    cell,
+    executionCount,
+    output,
+    language,
+}) => {
+    return (
+        <JupyterNotebookCellRenderTemplate
+            idColumnContent={`Out [${executionCount}]`}
+            bodyColumnContent={
+                <ReactMarkdown children={'**[' + output.name + ']**\n```\n ' + sourceLines(output.text) + '```'} />
+            }
+            bodyColumnStyle={{
+                padding: '0px 8px 0px 8px',
+                margin: '0px 0px',
+                backgroundColor: output.name === 'stderr' ? '#FDD' : '#DFD',
+                border: '1px solid #AAA',
+            }}
+        />
+    );
+};
+
+// -- renders a cell of type 'error'
+const JupyterNotebookOutputErrorCell: React.FC<JupyterNotebookOutputCellProps> = ({
+    cell,
+    executionCount,
+    output,
+    language,
+}) => {
+    return (
+        <JupyterNotebookCellRenderTemplate
+            idColumnContent={`In [${executionCount}]`}
+            bodyColumnContent={
+                <ReactMarkdown
+                    children={
+                        '**[' +
+                        output.evalue +
+                        ']**\n```\n ' +
+                        sourceLines(output.traceback).replace(
+                            /[\u001b\u009b][[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-ORZcf-nqry=><]/g, // strip out ansi control sequences common in output traces
+                            ''
+                        ) +
+                        '\n```'
+                    }
+                />
+            }
+            bodyColumnStyle={{
+                padding: '0px 8px 0px 8px',
+                margin: '0px 0px',
+                border: '1px solid #AAA',
+                backgroundColor: '#FDD',
+            }}
+        />
+    );
+};
+
+// -- renders various data cell types (html, png, text)
+const JupyterNotebookOutputDataCell: React.FC<JupyterNotebookOutputCellProps> = ({
+    cell,
+    executionCount,
+    output,
+    language,
+}) => {
+    return (
+        <JupyterNotebookCellRenderTemplate
+            idColumnContent={`Out [${executionCount}]`}
+            bodyColumnContent={
+                <Container
+                    style={{
+                        marginBottom: '0px',
+                        boxShadow: 'none',
+                    }}
+                >
+                    {output.data['text/html'] ? (
+                        parse(sourceLines(output.data['text/html']), { trim: true })
+                    ) : output.data['image/png'] ? (
+                        <img
+                            src={'data:image/png;base64,' + output.data['image/png'].replace(/\n/g, '')}
+                            alt={'notebook diagram'}
+                        />
+                    ) : output.data['text/plain'] ? (
+                        <ReactMarkdown children={'```\n' + sourceLines(output.data['text/plain']) + '\n```'} />
+                    ) : (
+                        <b>unknown</b>
+                    )}
```

Comment:
Same here it will be better to move this logic in separate component maybe

Reply:
perhaps, handling of these types is trickier - will consider in backlog